File: src/main/resources/output/IFC4X3_js/Entities/IfcAppliedValue/IfcAppliedValue.js
class IfcAppliedValue {
    constructor() {
        /** @type {IFCLABEL} */
        this.Name = null;
        /** @type {IFCTEXT} */
        this.Description = null;
        /** @type {IFCAPPLIEDVALUESELECT} */
        this.AppliedValue = null;
        /** @type {IFCMEASUREWITHUNIT} */
        this.UnitBasis = null;
        /** @type {IFCDATE} */
        this.ApplicableDate = null;
        /** @type {IFCDATE} */
        this.FixedUntilDate = null;
        /** @type {IFCLABEL} */
        this.Category = null;
        /** @type {IFCLABEL} */
        this.Condition = null;
        /** @type {IFCARITHMETICOPERATORENUM} */
        this.ArithmeticOperator = null;
        /** @type {IFCAPPLIEDVALUE[]} */
        this.Components = null;
    }

    // === EXTENDED BY ===
    // IfcCostValue

    // === INVERSE CLAUSES ===
    // HasExternalReference : SET [0:?] OF IfcExternalReferenceRelationship FOR RelatedResourceObjects
}

File: src/main/resources/output/IFC4X3_vbnet/Entities/IfcAppliedValue/IfcAppliedValue.vb
Public Class IfcAppliedValue
    Public Name As IfcLabel
    Public Description As IfcText
    Public AppliedValue As IfcAppliedValueSelect
    Public UnitBasis As IfcMeasureWithUnit
    Public ApplicableDate As IfcDate
    Public FixedUntilDate As IfcDate
    Public Category As IfcLabel
    Public Condition As IfcLabel
    Public ArithmeticOperator As IfcArithmeticOperatorEnum
    Public Components As List(Of IfcAppliedValue)

    ' === EXTENDED BY ===
    ' IfcCostValue

    ' === INVERSE CLAUSES ===
    ' HasExternalReference : SET [0:?] OF IfcExternalReferenceRelationship FOR RelatedResourceObjects
End Class
